moveit2.repos

```diff
@@ -11,24 +11,27 @@ repositories:
     type: git
     url: https://github.com/ros/joint_state_publisher
     version: ros2-devel
-  # TODO(henningkayser): Change to 'ros-planning/ros2' once https://github.com/ros-planning/random_numbers/pull/18 is merged
   random_numbers:
     type: git
-    url: https://github.com/PickNikRobotics/random_numbers
-    version: ros2-devel
-  # TODO(henningkayser): merge dependencies below into upstream packages
+    url: https://github.com/ros-planning/random_numbers
+    version: ros2
   geometric_shapes:
     type: git
-    url: https://github.com/PickNikRobotics/geometric_shapes
-    version: ros2
+    url: https://github.com/ros-planning/geometric_shapes
+    version: dashing-devel
+  geometry2:
+    type: git
+    url: https://github.com/ros2/geometry2
```

Comment:
@henningkayser I think we have fixed the `-Wcast-qual` warnings in the upstream repo, but I find colcon seems to always search the header files under `/opt` in priority even though the correct `setup.bash` are sourced.

Reply:
Yes, including the upstream repo should have fixed this, I'm not sure why colcon would prefer system installs. I assume that there is another system dependency that depends on `geometry2` leading to `opt` being prioritized. Export and import instructions seem correct. Since `geometry2` is a part of `ros-base` and therefore always installed in our docker container, I see no other other way than either switching to `ros-core` as base image or to enable `-Wcast-qual` until the fix is released.

Reply:
I think it's less intrusive to just apply your fix temporarily